poetry/console/commands/lock.py

```diff
@@ -23,6 +34,13 @@ def handle(self):
             self.io, self.env, self.poetry.package, self.poetry.locker, self.poetry.pool
         )
 
-        installer.lock()
+        if self.option("check"):
+            return (
+                0
+                if self.poetry.locker.is_locked() and self.poetry.locker.is_fresh()
+                else 1
+            )
+        else:
+            installer.lock()
 
-        return installer.run()
+            return installer.run()
```

Comment:
As there is already a return in the previous `if`, I think it would be better to remove this else. It will help with further additions to the command.

Reply:
Just to make sure I understand, you would prefer the following?

```suggestion
      if self.option("check"):
          return (expr)
          
      installer.lock()
      return installer.run()
```

Reply:
Yes exactly :+1: .

Reply:
Done!